### PR TITLE
Telemetry implementation

### PR DESCRIPTION
**Pull request by Jozefiel**

### CODE REVIEW

File: frinx/common/worker/worker.py [hunk 1]
```diff
@@ -102,22 +105,39 @@ def execute(self, task: Task) -> TaskResult:
...
             cls.WorkerInput.parse_obj(task['inputData'])
         except ValidationError as error:
             logger.error('Validation error occurred: %s', error)
             return TaskResult(status=TaskResultStatus.FAILED, logs=[TaskExecLog(str(error))]).dict()
+
         try:
-            # TODO check if ok
             logger.debug('Executing task %s:', task)
-            task_result = cls.execute(cls, Task(**task)).dict()  # type: ignore[arg-type]
+            task_result: TaskResult
```
Comment (dav1d-b) [inline frinx/common/worker/worker.py]: Do we need this line?

**Review by dav1d-b — COMMENTED** [review]
Also I am not really sure if we need the singleton pattern for this. Maybe we could just initialize the Metrics class inside WorkerImpl?

File: frinx/common/worker/worker.py (hunk 1 above)
Comment (Jozefiel) [reply frinx/common/worker/worker.py]: True, removed

**Comment by Jozefiel** [comment]
Probably yes, because we want to introduce also other metrics into the SDK.
Especially to the conductor client.

**Comment by dav1d-b** [comment]
I thought that singleton could be useful if we would use the class similarly to a logger, but I don't think it will be possible. The implementation is not general, so we cannot use it in arbitrary piece of code.

File: frinx/common/telemetry/metrics.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                self.counters[name] = self.__generate_counter(name, documentation, label_names)
+            return self.counters[name]
+
+    def __get_gauge(
+            self, name: MetricName, documentation: MetricDocumentation, label_names: list[MetricLabel]
+    ) -> Gauge:
+        if self.settings.metrics_enabled:
+            if name not in self.gauges:
+                self.gauges[name] = self.__generate_gauge(name, documentation, label_names)
+            return self.gauges[name]
```
Comment (dav1d-b) [inline frinx/common/telemetry/metrics.py]: This function can return None if metrics are not enabled. So the proper return type should be Optional[Gauge]. Btw Is mypy check passing?

File: frinx/common/worker/worker.py [hunk 3]
```diff
@@ -102,22 +109,39 @@ def execute(self, task: Task) -> TaskResult:
...
+    def _execute_func(cls, task: RawTaskIO) -> TaskResult:
+        if not metrics.settings.metrics_enabled:
+            return cls.execute(cls, Task(**task)).dict()  # type: ignore[arg-type]
+
+        start_time = time.time()
+        task_result: TaskResult = cls.execute(cls, Task(**task)).dict()  # type: ignore[arg-type]
+        finish_time = time.time()
+        record_task_execute_time(metrics, task.get('taskType'), finish_time - start_time)
+        return task_result
+
```
Comment (dav1d-b) [inline frinx/common/worker/worker.py:144]: I was thinking again about this. Maybe a set of decorators would fit here better. But its just an idea.

**Review by dav1d-b — APPROVED** [review]

**Pull request merged**